CodeStarterCamp_Week4/main.swift

```diff
@@ -80,52 +86,166 @@ class BodyCondition {
     }
 }
 
+func checkInputCount (inputCountRoutines: String?) throws -> Int {
+    var returnInt = 0
+    if let inputToString = inputCountRoutines {
+        if let inputToInt = Int(inputToString) {
+            guard inputToInt >= 1 && inputToInt < 100 else {
+                throw ActivityError.invalidInput
+            }
+            returnInt = inputToInt
+        } else {
+            throw ActivityError.invalidInput
+        }
+    } else {
+        throw ActivityError.invalidInput
+    }
+    
+    return returnInt
+}
```

Comment:
입력값의 오류를 검사해주는 메서드네요 👍 이렇게 중첩적으로 옵셔널 바인딩을 해야한다면 

```suggestion
func checkInputCount (inputCountRoutines: String?) throws -> Int {
    var returnInt = 0
    if let inputToString = inputCountRoutines, let inputToInt = Int(inputToString) {
        guard inputToInt >= 1 && inputToInt < 100 else {
            throw ActivityError.invalidInput
        }
        returnInt = inputToInt
    } else {
        throw ActivityError.invalidInput
    }
    
    return returnInt
}
```

이런 식으로 표현해봐도 좋을 것 같아요 . 그럼 들여쓰기 한번이 줄어들어 가독성이 좀 더 좋아지겠죠?

Reply:
반영하였습니다 감사합니다 😄